RFS/src/main/java/com/rfs/common/OpenSearchClient.java

```diff
@@ -281,13 +260,40 @@ public Optional<ObjectNode> updateDocument(String indexName, String documentId,
             .retryWhen(Retry.backoff(3, Duration.ofSeconds(1)).maxBackoff(Duration.ofSeconds(10)))
             .block();
         if (response.code == HttpURLConnection.HTTP_OK) {
-            return Optional.of(body);
+            return Optional.of(newBody);
         } else {
             // The only response code that can end up here is HTTP_CONFLICT, as everything is an error above
             // This indicates that we didn't acquire the optimistic lock
             return Optional.empty();
         }
     }
+    
+    /*
+     * Update a document forcefully by skipping optimistic locking and overwriting the document regardless of versioning.
+     * Returns the new body used in the update.
+     */
+    public ObjectNode updateDocumentForceful(String indexName, String documentId, ObjectNode newBody) {
```

Comment:
Seems like this could just likely wrap `updateDocument(...)` by sending null instead of duplicated the logic

Reply:
The behavior seemed sufficiently different that it felt clearer to make a separate method.
* The optimistic locking logic at the start changes the REST Path
* The acceptable response codes are different
* The post-response handling is different

This could be solved with several additional `if` statements, but IMO I'd rather have two more-focused things than one more conditionally complex thing (at least, in this case).